core/EE_Registry.core.php

```diff
@@ -1517,20 +1583,25 @@ public function reset_model($model_name)
      *                                  code instead can just change the model context to a different blog id if
      *                                  necessary
      * @return EE_Registry
+     * @throws InvalidInterfaceException
+     * @throws InvalidDataTypeException
      * @throws EE_Error
      * @throws ReflectionException
+     * @throws InvalidArgumentException
      */
     public static function reset($hard = false, $reinstantiate = true, $reset_models = true)
     {
-        $instance = self::instance();
+        $instance            = self::instance();
         $instance->_cache_on = true;
         // reset some "special" classes
         EEH_Activation::reset();
-        $hard = apply_filters( 'FHEE__EE_Registry__reset__hard', $hard);
-        $instance->CFG = EE_Config::reset($hard, $reinstantiate);
-        $instance->CART = null;
-        $instance->MRM = null;
-        $instance->AssetsRegistry = $instance->create('EventEspresso\core\services\assets\Registry');
+        $hard                     = apply_filters('FHEE__EE_Registry__reset__hard', $hard);
```

Comment:
I wonder if it'd be better to label this variable `$do_hard_reset` so that it's clearer what it represents?

Reply:
huh... that must have gotten reformatted after I removed some debug code... didn't mean to adjust that line at all